.travis.yml

```diff
@@ -1,12 +1,14 @@
 sudo: false
-dist: bionic
+dist: focal
 language: python
 python:
   - 3.6
   - 3.8
 
 install:
   - pip install tox tox-travis
+  - sudo apt update
+  - sudo apt install -y libgirepository1.0-dev libnm0 python3-gi
```

Comment:
Why is an update needed? And these packages were not needed before?

Reply:
These are the requirements to get PyGObject building on Ubuntu but it does not seem to fix Travis. I suspect it might be missing a running NetworkManager. A proper solution would probably be to make Travis run a container like Nmstate itself does.

Reply:
So this is at least necessary to be able to install nmstate in Travis. So this improves the situation but does not fix it, yet. Since it moves into the right direction, I would be in favor or merging this to allow others to build upon this. Maybe mocking NM would be enough for the actual unit tests and we only need nmstate for the static code analysis.

Reply:
Ah, the apt update just refreshed the repository information it does not upgrade the system. On Debian update/upgrade are different things.